kolibri/plugins/setup_wizard/assets/src/actions.js

```diff
@@ -0,0 +1,24 @@
+const Kolibri = require('kolibri');
+
+const DeviceOwnerResource = Kolibri.resources.DeviceOwnerResource;
+const FacilityResource = Kolibri.resources.FacilityResource;
+
+function createDeviceOwnerAndFacility(store, deviceownerpayload, facilitypayload) {
+  const DeviceOwnerModel = DeviceOwnerResource.createModel(deviceownerpayload);
+  const deviceOwnerPromise = DeviceOwnerModel.save(deviceownerpayload);
+  const FacilityModel = FacilityResource.createModel(facilitypayload);
+  const facilityPromise = FacilityModel.save(facilitypayload);
+  const promises = [deviceOwnerPromise, facilityPromise];
+  Promise.all(promises).then(responses => {
+    // redirect to learn page after successfully created the DeviceOwner and Facility.
+    window.location = window.location.origin.concat('/learn/');
```

Comment:
to use reverse requires modify the api endpoint to response with the reversed url. or is there a way to reverse solely on front end?